File: CPP/Abr_14_Usando_MeuPacote_CPP/Usando_Aluno/Aluno.cpp
#include "Aluno.h"

string situacao(double freq, double nf) {
  if ((freq >= 75) && (nf >= 6.0))
     return("APROVADO");
  else if ((freq < 75.0) && (nf >= 6.0))
          return("REPROVADO POR FREQUÊNCIA");
       else if ((freq >= 75.0) && (nf < 6.0))
               return("REPROVADO POR NOTA");
            else return("REPROVADO POR NOTA E FREQUÊNCIA");
  }

string toString(string nome, double freq, double nf) {
  char result[200];

  sprintf(result, "Nome do Aluno: %s\n"
                  "Nota Final...: %.2f\n"
                  "Frequência...: %.2f%%\n"
                  "Situação.....: %s\n\n",
                  nome.c_str(),
                  nf,
                  freq,
                  situacao(freq, nf).c_str());

  return(result);
}

File: POO_2026_1/src/MeuPacote/Aluno.java
package MeuPacote;

public class Aluno {
  
  public static String situacao(double nf, double freq) {
    if ((nf >= 6.0) && (freq >= 75.0))
        return("<<< APROVADO >>>");
    else if ((nf >= 6.0) && (freq < 75.0))
            return("<<< REPROVADO POR FREQUÊNCIA >>>");
         else if ((nf < 6.0) && (freq >= 75.0))
                 return("<<< REPROVADO POR NOTA >>>");
              else return("<<< REPROVADO POR NOTA E FREQUÊNCIA >>>");
  }
  
  public static String toString(String nome, double nf, double freq) {
    String result;
    
    result = String.format("Nome do(a) Aluno(a): %s\n", nome) +
             String.format("Frequência.........: %.2f%%\n", freq) +
             String.format("Nota Final.........: %.2f\n", nf) +
             String.format("Situação...........: %s", situacao(nf, freq));
    
    return(result);
  }

}
